.github/workflows/docs.yaml

```diff
@@ -26,6 +26,10 @@ jobs:
       with:
         submodules: true
 
+    - uses: actions/setup-python@v5.3.0
+      with:
+        python-version: '3.12'
+
```

Comment:
You can just set the runner to ubuntu-24.04 to get python 3.12 pre-installed

Reply:
Ha! I assumed latest was the latest. Thank you, change applied. I also rebased to clean the history.